_assets/patches/geth/0027-geth-185-fix-race-in-downloader-cancel.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/eth/downloader/downloader.go b/eth/downloader/downloader.go
+index eb64f05a..3e8c9cb1 100644
+--- a/eth/downloader/downloader.go
++++ b/eth/downloader/downloader.go
+@@ -230,6 +230,7 @@ func New(mode SyncMode, stateDb ethdb.Database, mux *event.TypeMux, chain BlockC
+ 		},
+ 		trackStateReq: make(chan *stateReq),
+ 	}
++	dl.cancelWg.Add(1)  // Add a sentinel to avoid getting flagged as a race condition in case Cancel is called before spawnSync
+ 	dl.downloads.Add(1) // Add a sentinel value to avoid waiting with zero count and getting a race condition signaled when terminating Downloader before starting sync
+ 	go dl.qosTuner()
+ 	go dl.stateFetcher()
+@@ -527,6 +528,7 @@ func (d *Downloader) cancel() {
+ // finish before returning.
+ func (d *Downloader) Cancel() {
+ 	d.cancel()
++	go func() { d.cancelWg.Done() }() // Remove sentinel
```

Comment:
I have the same thought about yesterday, since `Done` just does `-1`, I would keep it in the current goroutine, but since we already did like this in the other patches it makes sense

Reply:
ah another discussion about it already started above, I'll follow that one